addons/jest/src/index.js

```diff
@@ -27,9 +28,27 @@ export const withTests = userOptions => {
   };
   const options = Object.assign({}, defaultOptions, userOptions);
 
-  return (...testFiles) => (storyFn, { kind, story }) => {
-    emitAddTests({ kind, story, testFiles, options });
+  return (...args) => {
+    if (typeof args[0] === 'string') {
+      return deprecate((story, { kind }) => {
+        emitAddTests({ kind, story, testFiles: args, options });
 
-    return storyFn();
+        return story();
+      }, 'Passing component filenames to the `@storybook/addon-jest` via `withTests` is deprecated. Instead, use the `jest` story parameter');
+    }
+
+    const [
+      story,
+      {
+        kind,
+        parameters: { jest: testFiles },
+      },
+    ] = args;
+
+    if (testFiles && !testFiles.skip) {
```

Comment:
Where does this `testFiles.skip` come from? Is this the same as the `disable` flag that we have in the other parameterized addons?

Reply:
Poking @tmeasday just in case you missed this comment, since this is part of the 4.0 release.

Reply:
Thanks @Keraito -- I have not forgotten, just getting around to this, sorry!